File: ApplicationCommon/DTOs/Account/JobSeekerAccountDTO.cs
using ApplicationCommon.DTOs.Address;
using ApplicationCommon.DTOs.Resume;
using ApplicationCommon.DTOs.User;
using ApplicationCommon.DTOs.UserJobInteractions;

namespace ApplicationCommon.DTOs.Account;

public class JobSeekerAccountDTO
{
  public  int Id { get; set; }
  
  public string FirstName { get; set; }
  
  public string LastName { get; set; }
  
  public string PhoneNumber { get; set; }
  
  public int UserId { get; set; }

  public UserDTO User { get; set; }
  
  public int AddressId { get; set; }
  public AddressDTO Address { get; set; }
  public ResumeDTO Resume { get; set; }
  public List<InterviewDTO> Interview { get; set; }
  public List<JobApplyDTO> JobApply { get; set; }
  public List<SavedJobDTO> SavedJob { get; set; }
}

File: HelperJobby/client-app/src/DTOs/accountDTOs/JobSeekerAccountDTO.ts
import {AddressDTO} from "../addressDTOs/AddressDTO";
import {ResumeDTO} from "../resumeRelatedDTOs/ResumeDTO";
import {JobApplyDTO} from "../userJobInteractionsDTOs/JobApplyDTO";
import {SavedJobDTO} from "../userJobInteractionsDTOs/SavedJobDTO";
import {InterviewDTO} from "../userJobInteractionsDTOs/InterviewDTO";
import {UserDTO} from "../userRelatedDTOs/UserDTO";

export interface JobSeekerAccountDTO {
    id: number;
    firstName: string;
    lastName: string;
    phoneNumber: string;
    userId: number;
    user: UserDTO;
    addressId: number | null;
    address: AddressDTO | null;
    resume: ResumeDTO | null;
    interviews: InterviewDTO[];
    jobApplies: JobApplyDTO[];
    savedJobs: SavedJobDTO[];
}